core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutorBuilder.java

```diff
@@ -0,0 +1,140 @@
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.Flags;
+
+/**
+ * Provides a builder that is useful for creating a {@link ScheduledExecutorService}.
+ *
+ * @see CommonPools#blockingTaskExecutor()
+ */
+public final class BlockingTaskExecutorBuilder {
+
+    private String threadNamePrefix = "armeria-blocking-tasks";
+    private int numThreads = Flags.numCommonBlockingTaskThreads();
+    private long keepAliveTimeMillis = 600;
+    private boolean daemon = true;
+    private int priority = Thread.NORM_PRIORITY;
+    private Function<? super Runnable, ? extends Runnable> taskFunction = Function.identity();
+
+    BlockingTaskExecutorBuilder() {}
+
+    /**
+     * Sets the prefix of thread names.
+     */
+    public BlockingTaskExecutorBuilder threadNamePrefix(String threadNamePrefix) {
+        requireNonNull(threadNamePrefix, "threadNamePrefix");
+        this.threadNamePrefix = threadNamePrefix;
+        return this;
+    }
+
+    /**
+     * Sets the number of blocking task executor threads.
+     */
+    public BlockingTaskExecutorBuilder numThreads(int numThreads) {
+        checkArgument(numThreads >= 0, "numThreads: %s (expected: >= 0)", numThreads);
+        this.numThreads = numThreads;
+        return this;
+    }
+
+    /**
+     * Sets the amount of keep alive time in seconds.
+     */
+    public BlockingTaskExecutorBuilder keepAliveTime(Duration keepAliveTime) {
+        checkArgument(!requireNonNull(keepAliveTime, "keepAliveTime").isNegative(),
+                      "keepAliveTime: %s (expected: >= 0)");
+        return keepAliveTimeMillis(keepAliveTime.toMillis());
+    }
+
+    /**
+     * Sets the amount of keep alive time in seconds.
+     */
+    public BlockingTaskExecutorBuilder keepAliveTimeMillis(long keepAliveTimeMillis) {
+        checkArgument(keepAliveTimeMillis >= 0, "keepAliveTimeMillis: %s (expected: >= 0)", keepAliveTimeMillis);
+        this.keepAliveTimeMillis = keepAliveTimeMillis;
+        return this;
+    }
+
+    /**
+     * Sets the flag of daemon for new threads.
+     */
+    public BlockingTaskExecutorBuilder daemon(boolean daemon) {
+        this.daemon = daemon;
+        return this;
+    }
+
+    /**
+     * Sets the priority for new threads.
+     */
+    public BlockingTaskExecutorBuilder priority(int priority) {
+        checkArgument(priority >= Thread.MIN_PRIORITY && priority <= Thread.MAX_PRIORITY,
+                      "priority: %s (expected: Thread.MIN_PRIORITY <= priority <= Thread.MAX_PRIORITY)",
+                      priority);
+
+        this.priority = priority;
+        return this;
+    }
+
+    /**
+     * Use this method to set additional work before or after the Runnable is run. For example:
+     * <pre>{@code
+     * ThreadFactories.builder("thread-prefix")
```

Comment:
```suggestion
     * BlockingTaskExecutor.builder("thread-prefix")
```

Reply:
The review seems not to be applied. 😄
Otherwise, were you intentionally use `ThreadFactories` here?

Reply:
Hmm.. ? it is not applied ? https://github.com/line/armeria/pull/3389/commits/2c38168f9ed888c4a57564f3b2180b96ec85fc18#diff-eb2970174ab206c5b370f1809070743d907b422aa871b8e8c1a712c470c727e7R107